packages/protocol/test-sol/governance/network/BlockchainParameters.t.Sol

```diff
@@ -0,0 +1,154 @@
+pragma solidity ^0.5.13;
+
+import "@celo-contracts/governance/BlockchainParameters.sol";
+
+import "celo-foundry/Test.sol";
+
+import { Constants } from "@test-sol/constants.sol";
+import { Utils } from "@test-sol/utils.sol";
+
+contract BlockchainParametersTest is Test, Constants, Utils {
+  uint256 constant EPOCH_SIZE = 100;
+  uint256 constant gasLimit = 7000000;
+  uint256 constant gasForNonGoldCurrencies = 50000;
+
+  BlockchainParameters blockchainParameters;
+
+  event IntrinsicGasForAlternativeFeeCurrencySet(uint256 gas);
+  event BlockGasLimitSet(uint256 limit);
+  event UptimeLookbackWindowSet(uint256 window, uint256 activationEpoch);
+  event OwnershipTransferred(address indexed previousOwner, address indexed newOwner);
+
+  function setUp() public {
+    ph.setEpochSize(EPOCH_SIZE);
+    blockchainParameters = new BlockchainParameters(true);
+  }
+
+  function test_setBlockGasLimit_set_variable() public {
+    blockchainParameters.setBlockGasLimit(gasLimit);
+    assertEq(blockchainParameters.blockGasLimit(), gasLimit);
+  }
+
+}
+
+contract BlockchainParametersTest_initialize is BlockchainParametersTest {
+  uint constant lookbackWindow = 20;
+
+  function test_ShouldSetTheVariables() public {
+    blockchainParameters.initialize(gasForNonGoldCurrencies, gasLimit, lookbackWindow);
+    assertEq(blockchainParameters.blockGasLimit(), gasLimit);
+    blockTravel(EPOCH_SIZE);
+    assertEq(blockchainParameters.getUptimeLookbackWindow(), lookbackWindow);
+  }
+
+  function test_ShoudEmitIntrinsicGasForAlternativeFeeCurrencySet() public {
+    vm.expectEmit(true, true, true, true);
+    emit IntrinsicGasForAlternativeFeeCurrencySet(gasForNonGoldCurrencies);
+    emit UptimeLookbackWindowSet(lookbackWindow, 2);
+    emit OwnershipTransferred(msg.sender, msg.sender);
+    blockchainParameters.initialize(gasForNonGoldCurrencies, gasLimit, lookbackWindow);
+  }
+
+}
+
+contract BlockchainParametersTest_setBlockGasLimit is BlockchainParametersTest {
+
+  function test_ShouldSetTheVariable () public {
+    blockchainParameters.setBlockGasLimit(gasLimit);
+    assertEq(blockchainParameters.blockGasLimit(), gasLimit);
+  }
+
+  function test_ShoudEmitBlockGasLimitSet () public {
+    vm.expectEmit(true, true, true, true);
+    emit BlockGasLimitSet(gasLimit);
+    blockchainParameters.setBlockGasLimit(gasLimit);
+  }
+
+  function test_ShouldRevert_WhenCalledByNonOwner() public {
+    vm.prank(msg.sender);
```

Comment:
NIT: to keep with the format we have been using, it might be a ggod Idea to create a `nonOwner` account to submit these transactions